lib/ftp/src/ftp.erl

```diff
@@ -120,7 +127,7 @@
 -type shortage_reason()  :: 'etnospc' | 'epnospc'.
 -type restriction_reason() :: 'epath' | 'efnamena' | 'elogin' | 'enotbinary'.
 -type common_reason() ::  'econn' | 'eclosed' | term().
--type file_write_error_reason() :: term(). % See file:write for more info
+-type file_write_error_reason() :: file:posix() | badarg | terminated. % See file:write for more info
```

Comment:
Isn't `badarg` something we associate with an exception? 
A bad type of argument can result in an exception while a correct type of argument can result in `ok | {error, Reason}`?

Reply:
in this case, since the previous code was referring to `file:write`, I simply placed those types in here